### PR TITLE
Add view for tracks, simplier tracks view, rework directives...

### PR DESCRIPTION
**Pull request by Pitros**
Okay, so it's big commit, It should be split into separate ones, but I changed files so bad that everything needs to be pushed at once.

Changes:
1. Added track view with comments #335 
2. Tracks view is now simplier #336
3. Rewrote `show more` button behavior
4. Improved `like` button behavior - now you can like and unlike tracks without reloading 
5. Liked tracks are now marked as liked
6. Smaller css fixes (followings view, container padding)
7. Submiting search by pressing enter now hides dropdown

Because it's one commit, if something needs to be removed or changed because You don't like it etc. I'll just change commit on my branch.

Screenshots:
image
image
image

### CODE REVIEW

**Comment by Chriz007** [comment]
Can you add a repost icon so people can tell who the track was reposted by?

**Comment by Pitros** [comment]
@Chriz007 that would require to use soundcloud api v2, which is undocumented and we probably shouldn't use it because it might violate TOS. I tried contacting soundcloud devs but they are pretty good at not responding.

**Comment by weblancaster** [comment]
This is big work @Pitros thanks a lot..

I would suggest not having the "active" state in the word "like" just the heart active is enough and more simple.. I don't want to have the orange state all over the App.

**Comment by weblancaster** [comment]
The tracks come with a key `user_favorite` can't you use that instead of make another call to know if it's favorited or not?

The "page refresh" was to replicate/reload just the liked items not just remove from the UI.

**Comment by Pitros** [comment]
@weblancaster i'll limit liked color to just heart as you wish.

The thing is, on `/me/activities/` soundcloud api for each track returns: `user_favorite: null`, probably because in API V2 there's no `user_favorite` at all. So on soundcloud.com they fetch all your likes and later check if track ID matches with id list from your likes list (as far as i checked theirs code). 
For other api calls as search, likes etc it returns `user_favorite` as true or false depending if user liked that track normally. So for `stream` i had to that workaround which is fast engouh.

About that refresh - you could like track but to unlike it you had to go to likes, but when you unliked wrong track it disapeared because of reload (I've lost 1 or 2 tracks by mistake that way).

**Comment by Pitros** [comment]
@weblancaster is that looking good for you?
image

Here's the second idea, but it's harder to say which are liked without closer look.
image

**Comment by Chriz007** [comment]
Oh hey @Pitros is there a feature to add your own comment? Or not at the moment?

**Comment by Pitros** [comment]
@Chriz007 it's not included yet.

File: app/public/js/common/SCapiService.js [hunk 1]
```diff
@@ -186,6 +186,86 @@ app.service('SCapiService', function($http, $window, $q, $log, $state, $statePar
...
+                // something went wrong
+                return $q.reject(response.data);
+            });
+    };
+
+    /**
+     * Responsible to get user favorite tracks ids
+     * @return {[object]} data response
+     */
+    this.getFavoritesIds = function(next_href) {
```
Comment (weblancaster) [inline app/public/js/common/SCapiService.js]: The API response comes with `user_favorite` so let's just use that.. if it's null or empty nothing will happen, no need to all that extra work/code.

File: app/public/js/common/SCapiService.js [hunk 2]
```diff
@@ -186,6 +186,86 @@ app.service('SCapiService', function($http, $window, $q, $log, $state, $statePar
     };
 
     /**
+     * Responsible to get track details
+     * @return {[object]} data response
+     */
+    this.getTrackDetails = function(songId) {
```
Comment (weblancaster) [inline app/public/js/common/SCapiService.js]: No need to add new methods to call the SC api... you can just use the get method passing the `endpoint` and/or `params` that should work fine.

e.g
```
SCapiService.get('tracks/123')

SCapiService.get('tracks/123/comments', 'linked_partitioning=1&limit=30')
```

File: app/public/js/common/SCapiService.js [hunk 3]
```diff
@@ -186,6 +186,86 @@ app.service('SCapiService', function($http, $window, $q, $log, $state, $statePar
...
+                return $q.reject(response.data);
+            });
+    };
+
+
+    /**
+     * Responsible to get comments for track
+     * @return {[object]} data response
+     */
+    this.getTrackComments = function(songId) {
```
Comment (weblancaster) [inline app/public/js/common/SCapiService.js]: same as https://github.com/Soundnode/soundnode-app/pull/340/files#r38952706

File: app/public/js/common/SCapiService.js (hunk 1 above)
Comment (Pitros) [inline app/public/js/common/SCapiService.js]: Exactly nothing will happen, track won't be marked as liked when it's rather important information about track (on stream view - that's where it's always null).
As well in future cached favs list can be used to show on player bar if track was already liked.
Few lines of code isn't much.

File: app/public/js/common/SCapiService.js (hunk 2 above)
Comment (Pitros) [inline app/public/js/common/SCapiService.js]: Can be done. If you want to remove not needed methods you can also remove: getFollowing, getProfile and probably isFollowing

**Comment by weblancaster** [comment]
@Pitros Thanks for removing the unnecessary calls to SC.

On the like subject.. I keep getting this error when after the App starts.
image

**Comment by Pitros** [comment]
That's weird, it works totally fine for me. Function markLikedTracks exists in your stremCtrl.js?

**Comment by weblancaster** [comment]
Yep..I checked out locally your PR.

**Comment by weblancaster** [comment]
But even though I still don't think that's the best way to mark them as `liked` must be a better way.

It would be much appreciated if you split the PR into this one where contains the track view and the directive rework and another PR that contains the mark liked code

**Comment by Pitros** [comment]
I'll split it later into two PR if you really need that. 

Anyway I don't think that there is easier way to check if track is liked on stream view (search and liked tracks views does not require that), the same method is used on soundcloud.com. But you can obviously try to find other solution maybe there is better way.

**Comment by weblancaster** [comment]
Ok that works.. I just want to merge the track work but continue to review/test them as two different PR which will make much easier and faster for me.

**Comment by Pitros** [comment]
Splitted.

**Comment by weblancaster** [comment]
great thanks @Pitros

**Pull request merged**